pkgs/native_assets_builder/test_data/transformer/hook/build.dart

```diff
@@ -0,0 +1,60 @@
+// Copyright (c) 2024, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'dart:io';
+
+import 'package:native_assets_cli/native_assets_cli.dart';
+import 'package:transformer/src/transform.dart';
+
+void main(List<String> arguments) async {
+  await build(arguments, (config, output) async {
+    final dataDirectory =
+        Directory.fromUri(config.packageRoot.resolve('data/'));
+    // If data are added, rerun hook.
+    output.addDependency(dataDirectory.uri);
+
+    var transformedFiles = 0;
+    var cachedFiles = 0;
+
+    await for (final sourceFile in dataDirectory.list()) {
+      if (sourceFile is! File) {
+        continue;
+      }
+
+      final sourceName =
+          sourceFile.uri.pathSegments.lastWhere((e) => e.isNotEmpty);
```

Comment:
In the old uri vs. Strings discussion - `package:path`  has nice methods to get the `basename` of a file

Reply:
vs `package:file` 😄